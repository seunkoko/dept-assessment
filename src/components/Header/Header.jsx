// eslint-disable-next-line
import React from 'react';
import { Link } from 'react-router-dom';

import ButtonViewCase from '../Button/ButtonViewCase';
import HeaderModal from '../Modals/HeaderModal';

import './Header.scss';

const Header = (props) => {
	return (
    <>
      <div className="container-fluid carousel-style">
        {/* Navigation */}
        <nav className="navbar fixed-top header-style scroll-style">
          <ul className="nav justify-content-between scroll-ul-style pl-0">
            <li className="nav-item pl-0"> 
              <Link className="nav-link" to="#">DEPT</Link>
            </li>
            
            <li className="nav-item">
              <Link
                className="nav-link menu-icon"
                to="#"
                data-toggle="modal"
                data-target="#headerModal"
              >
                <p className="mr-4">MENU</p>
                <div
                  className="bar-container"
                >
                  <div className="bar1"></div>
                  <div className="bar2"></div>
                </div>
              </Link>
            </li>
          </ul>
        </nav>

        <h1 className="carousel-text">WORK</h1>
      </div>
      <ButtonViewCase />

      {/* Modal */}
      <HeaderModal />
    </>
	);
};

export default Header;
