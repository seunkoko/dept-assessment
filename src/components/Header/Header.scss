@import "../../stylesheets/variables";

.header-style {
  @include header-style;
  @include padding-desktop;

  margin: 1% 3%;

  > ul {
    @include underline;

    width: 100%;

    > li {
      vertical-align: bottom;
      width: fit-content;

      > a {
        color: $primary-black;
      }

      > .menu-icon {
        font-family: 'Teko-Light';
        font-size: 20px;
        display: inline-block;
        z-index: 1000000;

        > p {
          display: inline;
        }

        > a {
          margin-bottom: 0;
          height: fit-content;
        }
      }

      @media (max-width: $phone-width) {
        display: inline;
      }
    }

    @media (max-width: $phone-width) {
      border: none;
    }
  }

  @media (max-width: $desktop-width) {
    @include padding-laptop;

    margin: 2% 6%;
  }

  @media (max-width: $laptop-width) {
    @include padding-mobile;

    margin: 2% 3%;
  }

  @media (max-width: $phone-width) {
    @include padding-phone;
  }
}

.carousel-text {
  font-family: 'Teko-Regular';
  font-size: $banner-font;
  color: $primary-black;

  @media (max-width: $phone-width) {
    margin: auto;
    text-align: center;
    font-size: $phone-banner-font;
  }
}

.carousel-style {
  height: 600px;
  width: inherit;
  margin: 0;
  background: url("../../assets/images/florensis.png");
  background-attachment: fixed;
  background-size: 160%;
  background-repeat: no-repeat;
  background-position: top left;
  -webkit-background-size: 160%;
  -moz-background-size: 160%;
  -o-background-size: 160%;
  margin-top: 2%;
  padding: 10% 10%;

  @media (max-width: $phone-width) {
    padding-top: 20%;
    height: 350px;
    margin-top: 12%;
  }
  
  @media (max-width: $small-phone-width) {
    margin-top: 20%;
	}
}

.scroll-styles {
  margin: 0;
  background-color: #fff;

  > ul {
    border: none;
  }
}
