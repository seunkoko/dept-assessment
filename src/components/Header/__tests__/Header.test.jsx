import React from 'react';

import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header', () => {
	const component = shallow(<Header />);

	it('should render Header component', () => {
		expect(component).toMatchSnapshot();
  });
  
});