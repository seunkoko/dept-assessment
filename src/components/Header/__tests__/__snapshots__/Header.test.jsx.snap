// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<Fragment>
  <div
    className="container-fluid carousel-style"
  >
    <nav
      className="navbar fixed-top header-style scroll-style"
    >
      <ul
        className="nav justify-content-between scroll-ul-style pl-0"
      >
        <li
          className="nav-item pl-0"
        >
          <Link
            className="nav-link"
            to="#"
          >
            DEPT
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link menu-icon"
            data-target="#headerModal"
            data-toggle="modal"
            to="#"
          >
            <p
              className="mr-4"
            >
              MENU
            </p>
            <div
              className="bar-container"
            >
              <div
                className="bar1"
              />
              <div
                className="bar2"
              />
            </div>
          </Link>
        </li>
      </ul>
    </nav>
    <h1
      className="carousel-text"
    >
      WORK
    </h1>
  </div>
  <ButtonViewCase />
  <HeaderModal />
</Fragment>
`;
